packages/@react-aria/selection/src/useSelectableCollection.ts

```diff
@@ -369,7 +373,8 @@ export function useSelectableCollection(options: AriaSelectableCollectionOptions
 
     if (manager.focusedKey != null && scrollRef.current) {
       // Refocus and scroll the focused item into view if it exists within the scrollable region.
-      let element = scrollRef.current.querySelector(`[data-key="${CSS.escape(manager.focusedKey.toString())}"]`) as HTMLElement;
+      let nodeKey = idScope ? `${idScope}-${manager.focusedKey.toString()}` : manager.focusedKey.toString();
```

Comment:
One other place I think you may need to apply this change as well is in this function: https://github.com/adobe/react-spectrum/blob/50c7ada5d1880a174b6b6d3f43e8d90ee9bd4ad8/packages/%40react-aria/dnd/src/ListDropTargetDelegate.ts#L102-L108. The elementMap uses the data-keys on the nodes which are now prefixed: https://github.com/adobe/react-spectrum/blob/50c7ada5d1880a174b6b6d3f43e8d90ee9bd4ad8/packages/%40react-aria/dnd/src/ListDropTargetDelegate.ts#L90-L97 but then we use the collection keys (aka non prefixed keys) to access the items in the elementMap on line 108.

Reply:
Thanks, sometimes I wish test suite would cover all these 😓 .

As mentioned above, there is an edge case in here where a nested collection could look like this:

```tsx
<GridList id="grid" data-key="grid">
  <GridListItem id="dog" data-key="grid:dog"/>
  <GridListItem id="cat" data-key="grid:cat">
    <GridList id="grid:dog" data-key="grid:dog">
      <GridListItem id="dog" data-key="grid:dog:dog" />
    </GridList>
  </GridListItem>
</GridList>
```

causing the drop target to be incorrectly determined as the nested collection.

Reply:
Yeahhhh, I think we are going to need to expand the coverage of the RAC tests, I think we were hoping the tests in RSP would be redundant enough since they leverage the same hooks, but the different Collections API is proving that wrong. 

As for the edge case checking for role could certainly help. I think in general that the ListDropTarget delegate and other drag and drop logic will need to be updated for nested collection support anyways. Would changing the `idScope` from matching the user provided `id` to an autogenerated one help reduce the likelihood of this edge case as well? Reusing the `id` as the `idScope` feels more of a convenience and is certainly helpful for the user if they wanted to use `data-key` as a targeting strategy, but I'd say its not really meant as public API.